src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java

```diff
@@ -41,21 +41,23 @@
 import java.nio.file.Files;
 import java.nio.file.NoSuchFileException;
 import java.util.function.Function;
-import java.util.function.Supplier;
 import java.util.function.UnaryOperator;
 import org.apache.commons.io.IOUtils;
-import org.kohsuke.accmod.Restricted;
-import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
  * Utility class with the common logic to FileContent and UnfilteredFileContent.
  *
  * @author Stephen Connolly, M Ramón León
  */
-@Restricted(NoExternalUse.class)
-class BaseFileContent {
+final class BaseFileContent {
+
+    @FunctionalInterface
+    interface InputStreamSupplier {
+        InputStream get() throws IOException;
```

Comment:
Letting the caller handle the `IOException`. #174 appears to have been catching `FileNotFoundException` for this reason, but that could not actually have been thrown from the body of the `try` block: _other_ `IOException`s from handling `OutputStream`, perhaps, but the `Supplier` permitted no `IOException` so `FileNotFoundException` was being caught earlier.